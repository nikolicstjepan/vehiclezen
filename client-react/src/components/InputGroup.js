import React from "react";
import { capitalize } from "../lib/utils";

export default function InputGroup({ type = "text", name, value, onChange }) {
  return (
    <div className="mb-3">
      <label htmlFor={name} className="form-label">
        {capitalize(name)}
      </label>
      <input
        type={type}
        value={value}
        required
        className="form-control"
        name={name}
        id={name}
        onChange={onChange}
      />
    </div>
  );
}
