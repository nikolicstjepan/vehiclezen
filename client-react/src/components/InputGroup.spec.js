import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import InputGroup from "./InputGroup";

describe("InputGroup", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<InputGroup name="test" onChange={() => {}} value="value" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders input with default type text", () => {
    const component = shallow(<InputGroup name="test" onChange={() => {}} value="value" />);
    expect(component.find("input").prop("type")).toBe("text");
  });

  test("renders input with given value", () => {
    const component = shallow(<InputGroup name="test" onChange={() => {}} value="value" />);
    expect(component.find("input").prop("value")).toBe("value");
  });

  test("renders input with given name", () => {
    const component = shallow(<InputGroup name="test" onChange={() => {}} value="value" />);
    expect(component.find("input").prop("name")).toBe("test");
  });

  test("renders input with name as id", () => {
    const component = shallow(<InputGroup name="test" onChange={() => {}} value="value" />);
    expect(component.find("input").prop("id")).toBe("test");
  });

  test("renders label as capitalized name", () => {
    const component = shallow(<InputGroup name="test" onChange={() => {}} value="value" />);
    expect(component.find("label").text()).toBe("Test");
  });

  test("calls onChange when changed", () => {
    const onChange = jest.fn(() => {});
    const component = shallow(<InputGroup name="test" onChange={onChange} value="value" />);
    component.find("input").simulate("change", { target: { name: "test", value: "1" } });

    expect(onChange).toHaveBeenCalled();
  });

  test("calls onChange with new value", () => {
    const onChange = jest.fn(() => {});
    const change = { target: { name: "test", value: "1" } };

    const component = shallow(<InputGroup name="test" onChange={onChange} value="value" />);
    component.find("input").simulate("change", change);

    expect(onChange).toHaveBeenCalledWith(change);
  });
});
