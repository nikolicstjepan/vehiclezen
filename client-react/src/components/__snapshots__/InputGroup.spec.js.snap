// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup snapshot renders 1`] = `
<div
  className="mb-3"
>
  <label
    className="form-label"
    htmlFor="test"
  >
    Test
  </label>
  <input
    className="form-control"
    id="test"
    name="test"
    onChange={[Function]}
    required={true}
    type="text"
    value="value"
  />
</div>
`;
